components/automatic-points-table/commons/automatic-points-table.lua

```diff
@@ -298,4 +299,90 @@ function AutomaticPointsTable:addPositionData(pointsData)
 	)
 end
 
+function AutomaticPointsTable:generatePointsTable(pointsData, tournaments)
+	local columnCount = self:countColumns(tournaments)
+	local headerRow = self:generateHeaderRow(tournaments)
+
+	local divTable = DivTable.create() :row(headerRow)
+	divTable.root :addClass('border-color-grey') :addClass('border-bottom')
+
+	-- for top and left borders
+	local divWrapper = mw.html.create('div') :addClass('fixed-size-table-container')
+		:addClass('border-color-grey')
+		:css('width', tostring(450 + (columnCount * 50)) .. 'px')
+		:node(divTable:create())
+
+	-- for mobile / responsive scrolling
+	local responsiveWrapper = mw.html.create('div') :addClass('table-responsive')
+		:addClass('automatic-points-table')
+		:node(divWrapper)
+
+	return responsiveWrapper
+end
+
+function AutomaticPointsTable:countColumns(tournaments)
+	local columnCount = 0
+	-- tournaments with deductions have 2 columns
+	Table.iter.forEach(tournaments, function(t)
+		columnCount = columnCount + (t.shouldDeductionsBeVisible and 2 or 1)
+	end)
+	return columnCount
+end
+
+function AutomaticPointsTable:generateHeaderRow(tournaments)
+	local headerRow = DivTable.HeaderRow()
+	headerRow.root :addClass('diagonal')
+	-- fixed headers
+	local headers = {{
+		text = 'Ranking',
+		additionalClass = 'ranking'}, {
+		text = 'Team',
+		additionalClass = 'team' }, {
+		text = 'Total Points'
+	}}
+	Table.iter.forEach(headers,
+		function(header)
+			headerRow:cell(self:createHeaderCell(header))
+		end
+	)
+	-- variable headers (according to tournaments in given in module arguments)
+	Table.iter.forEach(tournaments,
+		function(tournament)
+			headerRow:cell(self:createHeaderCell({
+				text = tournament.display and tournament.display or tournament.name
+			}))
+
+			if tournament.shouldDeductionsBeVisible == true then
+				local deductionsHeader = tournament['deductionsheader']
+				headerRow:cell(self:createHeaderCell({
+					text = deductionsHeader and deductionsHeader or 'Deductions'
+				}))
+			end
+		end
+	)
+
+	return headerRow
+end
+
+function AutomaticPointsTable:createHeaderCell(header)
+	local additionalClass = header.additionalClass
+
+	local innerDiv = self:divWrap(header.text)
+		:addClass('border-color-grey')
+		:addClass('content')
+
+	local outerDiv = mw.html.create('div')
+		:addClass('diagonal-header-div-cell')
+	if additionalClass ~= nil then
+		outerDiv:addClass(additionalClass)
+	end
+	outerDiv:node(innerDiv)
+	return outerDiv
+end
+
+--- syntactic sugar

```

Comment:
Remove

Reply:
Sorry, what do I need to remove exactly?

Reply:
The comment